crates/matrix-sdk-ui/tests/integration/timeline/thread.rs

```diff
@@ -97,12 +97,32 @@ async fn test_thread_backpagination() {
         .await;
 
     let batch1 = vec![
-        factory.text_msg("Threaded event 4").event_id(event_id!("$3")).into_raw_sync().cast(),
-        factory.text_msg("Threaded event 3").event_id(event_id!("$4")).into_raw_sync().cast(),
+        factory
+            .text_msg("Threaded event 4")
+            .event_id(event_id!("$3"))
+            .in_thread(&thread_root_event_id, event_id!("$latest"))
```

Comment:
Fwiw, the second event ids are all incorrect, as this is the one used for the reply fallback, it should be set to the previous in-thread event, each time.